pkg/detector/detector.go

```diff
@@ -698,17 +699,27 @@ func (d *ResourceDetector) GetUnstructuredObject(objectKey keys.ClusterWideKey)
 }
 
 // ClaimPolicyForObject set policy identifier which the object associated with.
-func (d *ResourceDetector) ClaimPolicyForObject(object *unstructured.Unstructured, policy *policyv1alpha1.PropagationPolicy) (string, error) {
+func (d *ResourceDetector) ClaimPolicyForObject(object *unstructured.Unstructured, policy *policyv1alpha1.PropagationPolicy, resourceChangeByKarmada bool) (string, error) {
 	policyID := policy.Labels[policyv1alpha1.PropagationPolicyPermanentIDLabel]
-
+	generation := strconv.FormatInt(policy.Generation, 10)
 	objLabels := object.GetLabels()
+	samePolicy := false
 	if objLabels == nil {
 		objLabels = make(map[string]string)
 	} else if len(objLabels) > 0 {
 		// object has been claimed, don't need to claim again
 		if !excludeClusterPolicy(objLabels) &&
 			objLabels[policyv1alpha1.PropagationPolicyPermanentIDLabel] == policyID {
-			return policyID, nil
+			samePolicy = true
+			if _, exist := object.GetAnnotations()[policyv1alpha1.PropagationPolicyGenerationAnnotation]; !exist &&
+				!util.IsLazyActivationEnabled(policy.Spec.ActivationPreference) {
```

Comment:
If a lazy policy change `ActivationPreference` from `lazy` to `''`, it seems that the annotation still keep exist, in that case , maybe we should refresh cliamedGeneration all the time?

Reply:
my fault, you deleted the annotation when ActivationPreference is not lazy. Then it's fine here.